airbyte-webapp/src/components/connectorBuilder/Builder/BuilderField.tsx

```diff
@@ -53,25 +52,7 @@ const ArrayField: React.FC<ArrayFieldProps> = ({ name, value, setValue, error })
 };
 
 export const BuilderField: React.FC<BuilderFieldProps> = ({ path, label, tooltip, optional = false, ...props }) => {
-  let yupSchema = props.type === "array" ? yup.array().of(yup.string()) : yup.string();
-  if (!optional) {
-    yupSchema = yupSchema.required("form.empty.error");
-  }
-  const fieldConfig = {
-    name: path,
-    validate: (value: string) => {
-      try {
-        yupSchema.validateSync(value);
-        return undefined;
-      } catch (err) {
-        if (err instanceof yup.ValidationError) {
-          return err.errors.join(", ");
-        }
-        throw err;
-      }
-    },
-  };
-  const [field, meta, helpers] = useField(fieldConfig);
```

Comment:
I could no longer have BuilderFields dynamically set their own yup validation schemas, because this meant that switching away from some stream/global view would cause those validation schemas to no longer be applied, so errors across the form could not be tracked.

I instead had to define a single global validation schema that is used for the whole builder form, which I added to types.ts. This will make it a bit more annoying to add new fields, as that validation schema will also need to be updated, but I feel it is worth it for the better error tracking functionality.

(this is one more point in favor of generating these things from the JSON schema)